### PR TITLE
Added transform tutorial

### PR DESCRIPTION
**Pull request by thomelane**
## Description ##
Added a new tutorial on data transforms.
Covers `ToTensor`, `Compose`, `Resize` and `CenterCrop`.
All **random** transforms will be covered in the data augmentation tutorial.

### CODE REVIEW

**Comment by thomelane** [comment]
@NRauschmayr can you review when you get a chance, thanks!

**Comment by roywei** [comment]
Thanks for the contribution! Please remember to add it to nightly test, index.md, and run test tutorials once after finalized the content and before merging. Thanks!
@mxnet-label-bot add[Docs, Gluon]

File: docs/tutorials/gluon/transforms.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+### Using `ToTensor` and `.transform_first`
+
+Ordering of dimensions (sometimes called the data layout) is important for correct usage of a neural network. Currently our samples are ordered (height, width, channel) but we need to change this to (channel, height, width) before passing to our network. We also need to change our data type. Currently it's `uint8`, but we need to change this to `float32`.
+
+MXNet Gluon provides a number of useful `transform`s for common computer vision cases like this. We will use [`ToTensor`](https://mxnet.incubator.apache.org/api/python/gluon/data.html?highlight=totens#mxnet.gluon.data.vision.transforms.ToTensor to change the data layout and convert integers (between 0 and 255) to floats (between 0 and 1). We apply the transform to our `dataset` using the [`transform_first`](https://mxnet.incubator.apache.org/api/python/gluon/data.html?highlight=transform_first#mxnet.gluon.data.Dataset.transform_first) method. We have 2 elements per sample here (i.e. data and label), so the transform is only applied to the first element (i.e. data).
```
Comment (NRauschmayr) [inline docs/tutorials/gluon/transforms.md]: The link  [`ToTensor` ](...) is not correctly displayed.

Comment (NRauschmayr) [inline docs/tutorials/gluon/transforms.md]: `transform`s --> `transform` functions

File: docs/tutorials/gluon/transforms.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+print("data range: {} to {}".format(sample_data.min().asscalar(),
+                                    sample_data.max().asscalar()))
+print("label: {}".format(sample_label))
+```
+
+Our data has changed, while the label has been left untouched.
+
+### `Normalize`
+
+We scaled the values of our data samples between 0 and 1 as part of [`ToTensor`](https://mxnet.incubator.apache.org/api/python/gluon/data.html?highlight=totens#mxnet.gluon.data.vision.transforms.ToTensor but we may want or need to normalize our data instead: i.e. shift to zero-center and scale to unit variance. You can do this with the following steps:
```
Comment (NRauschmayr) [inline docs/tutorials/gluon/transforms.md]: Links is not correctly displayed

File: docs/tutorials/gluon/transforms.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+sample_data, sample_label = dataset[sample_idx]
+print("data range: {} to {}".format(sample_data.min().asscalar(),
+                                    sample_data.max().asscalar()))
+```
+
+### `Compose`
+
+We've now seen two examples of transforms: [`ToTensor`](https://mxnet.incubator.apache.org/api/python/gluon/data.html?highlight=totens#mxnet.gluon.data.vision.transforms.ToTensor and `Normalize`. We applied both transforms to our `dataset` through repeated calls to `transform_first`, but Gluon has a dedicated transform to stack other transforms that's preferred. With [`Compose`](https://mxnet.incubator.apache.org/api/python/gluon/data.html?highlight=compose#mxnet.gluon.data.vision.transforms.Compose) we can choose and order the transforms we want to apply.
+
+Caution: ordering of transforms is important. e.g. [`ToTensor`](https://mxnet.incubator.apache.org/api/python/gluon/data.html?highlight=totens#mxnet.gluon.data.vision.transforms.ToTensor) should be applied before `Normalize`, but after `Resize` and `CenterCrop`.
```
Comment (NRauschmayr) [inline docs/tutorials/gluon/transforms.md]: This is very important information! Are there more transforms where the order matters?

File: docs/tutorials/gluon/transforms.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+transform_fn = mx.gluon.data.vision.transforms.Resize(size=(10, 10))
+#transform_fn = mx.gluon.data.vision.transforms.CenterCrop(size=(10, 10))
+dataset = dataset.transform_first(transform_fn)
+sample_data, sample_label = dataset[sample_idx]
+plt.imshow(sample_data.asnumpy())
+```
+
+### Augmentation
+
+Augmenting samples is also common practice: that is, randomly permuting existing samples to make new samples to reduce issues of network overfitting. Using image samples as an example, you could crop to random regions, flip from left to right or even jitter the brightness of the image. Gluon has a number of random transforms that are covered in depth in the Data Augmentation tutorial.
```
Comment (NRauschmayr) [inline docs/tutorials/gluon/transforms.md]: I would add a link to this tutorial: https://mxnet.incubator.apache.org/versions/master/tutorials/python/types_of_data_augmentation.html

File: docs/tutorials/gluon/transforms.md (hunk 1 above)
Comment (thomelane) [reply docs/tutorials/gluon/transforms.md]: thanks

Comment (thomelane) [reply docs/tutorials/gluon/transforms.md]: removed the ``

File: docs/tutorials/gluon/transforms.md (hunk 2 above)
Comment (thomelane) [reply docs/tutorials/gluon/transforms.md]: thanks

File: docs/tutorials/gluon/transforms.md (hunk 3 above)
Comment (thomelane) [reply docs/tutorials/gluon/transforms.md]: Will add a note to augmentation tutorial about this too.

File: docs/tutorials/gluon/transforms.md (hunk 4 above)
Comment (thomelane) [reply docs/tutorials/gluon/transforms.md]: Going to replace this tutorial actually, so will wait till the new one is commited.

File: docs/tutorials/gluon/transforms.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+### `CenterCrop` and `Resize`
+
+Some networks require an input of a certain size (e.g. convolutional neural network with a final dense layer). Specifying the `size`, you can use `CenterCrop` and `Resize` to modify the spatial dimensions of an image. We show an example of downsampling to 10px by 10px here.
+
+
+```python
+dataset = mx.gluon.data.vision.CIFAR10()
+transform_fn = mx.gluon.data.vision.transforms.Resize(size=(10, 10))
+#transform_fn = mx.gluon.data.vision.transforms.CenterCrop(size=(10, 10))
```
Comment (ThomasDelteil) [inline docs/tutorials/gluon/transforms.md]: Why is this commented out?

Comment (thomelane) [reply docs/tutorials/gluon/transforms.md]: Was meant to be an alternative to resize for 10x10 output, but not clear so I have removed.
